src/execution_plan/ops/op_unwind.c

```diff
@@ -40,15 +44,25 @@ OpBase *NewUnwindOp(const ExecutionPlan *plan, AR_ExpNode *exp) {
  * if expression did not returned a list type value, creates a list with that value. */
 static void _initList(OpUnwind *op) {
 	op->list = SI_NullVal(); // Null-set the list value to avoid memory errors if evaluation fails.
-	SIValue new_list = AR_EXP_Evaluate(op->exp, op->currentRecord);
-	if(SI_TYPE(new_list) == T_ARRAY || SI_TYPE(new_list) == T_NULL) {
-		// Update the list value.
-		op->list = new_list;
+	if(AR_EXP_IsOperation(op->exp) && strcmp(op->exp->op.f->name, "range") == 0) {
+		op->from = AR_EXP_Evaluate(op->exp->op.children[0], op->currentRecord);
+		op->to = AR_EXP_Evaluate(op->exp->op.children[1], op->currentRecord);
+		if(op->exp->op.child_count == 3) {
+			op->step = AR_EXP_Evaluate(op->exp->op.children[2], op->currentRecord);
+		}
+		op->current = op->from;
```

Comment:
What will be the value of `op->current` when the range is empty?
`UNWIND range(0, 0)` ? similar to `UNWIND []`

Reply:
range(0, 0) is [0]